### ISSUE TITLE
What the-

### ISSUE THREAD

**Issue by CalibriLight** [HTV04/funkin-rewritten#14]
![Screenshot_2021-09-17-20-15-37](https://user-images.githubusercontent.com/59058076/133879096-9457ac87-a3cd-40a3-bd8c-20019092de13.jpg)
Android

**Comment by HTV04** [comment]
Might need to use a different compression format for Android devices, currently working on an ETC1 variant for all graphics

**Comment by CalibriLight** [comment]
> Yikes, might need to use a different compression format for Android devices, currently working on an ETC1 variant for all graphics
> 
> It's worth noting that Android isn't one of the officially supported platforms right now

Sad, can i put my bf image? If i can tell me how plz

**Comment by AM7999** [comment]
Just to add a somewhat "plausible" closure to your issue. As HTV04 has stated there will need to be a ETC1 variant of all the sprites, the reason i can think of him going this route is most android devices may not have enough VRAM   _(note this is not something you can adjust on your phone)_  and secondly _(unrelated to the first point of HTV04 saying you need ETC1)_  Even if you can plug in a USB keyboard into your phone using adapters and whatnot it wouldn't be the most  _(how do i say this)_  elegant solution as i personally think that having touch buttons corresponding to the note would make the game a bit more "accessible" to the average user who would probably not have the adapters to plug in a USB keyboard.

And finally You also last asked

>Sad, can i put my bf image? If i can tell me how plz

One way you could probably do it is to download and install Android Studio if you have a computer and figure out how to export images to ETC1

**Comment by GalaxytedYT** [comment]
Hold up, I have that same exact issue on my Raspberry Pi, I hadn't read this until now

**Comment by AM7999** [comment]
> Hold up, I have that same exact issue on my Raspberry Pi, I hadn't read this until now

@GalaxytedYT The reason for this is because Raspberry pis have similar hardware to a modern smartphone (arm SoC (system on a chip))

**Comment by GalaxytedYT** [comment]
> > Hold up, I have that same exact issue on my Raspberry Pi, I hadn't read this until now
> 
> @GalaxytedYT The reason for this is because Raspberry pis have similar hardware to a modern smartphone (arm SoC (system on a chip))
@AM7999 Then how come the Switch doesn't have that issue? I've heard it's an SoC as well, also I don't really mind the issue just as long as the songs work.

**Comment by HTV04** [comment]
> > > Hold up, I have that same exact issue on my Raspberry Pi, I hadn't read this until now
> > 
> > 
> > @GalaxytedYT The reason for this is because Raspberry pis have similar hardware to a modern smartphone (arm SoC (system on a chip))
> 
> @AM7999 Then how come the Switch doesn't have that issue? I've heard it's an SoC as well, also I don't really mind the issue just as long as the songs work.

It has a Tegra X1 SoC with more RAM and support for compressed textures, while the Raspberry Pi has a much less powerful SoC.

**Comment by Fordi** [comment]
I'm attempting to in-place (in the most recent beta release .love file) add a `pkm` imageType and set it as an option in the `hardwareCompression` setting.  Will report back later.

**Comment by AM7999** [comment]
Fun Fact: i have already gone ahead and done this whole dance with `etc1tool` it did not work as the textures _still_ did not show up

**Comment by Fordi** [comment]
Yeah, I think it's more an issue with löve being unable to decompress the whole image into memory (which, as I noted for the GF, is like 256M).  It fails to allocate and just displays whatever.

**Comment by AM7999** [comment]
i _have_ seen a "better optimized for the pi4/400" version of Löve. Not 100% sure that would change anything

**Comment by Fordi** [comment]
I have an 8GB Pi4 laying around somewhere.  I'm going to try running it on that.  May take a couple days before I get a chance though; it's almost midnight, and I have a day job.

**Comment by AM7999** [comment]
I have a pi4 w/ 8gb can confirm to you that it does practically the same thing


_actual fun fact time: i use that pi4 as my daily driver since my actual x86 based machine died_

**Comment by AM7999** [comment]
@HTV04 Do you mind changing this issue name to `Bad compression format for Android and the Raspberry pi 4`?

**Comment by AM7999** [comment]
> I have an 8GB Pi4 laying around somewhere. I'm going to try running it on that. May take a couple days before I get a chance though; it's almost midnight, and I have a day job.
> 
> If it runs, it means we're hitting the memory limit at 4 GB.
> 
> Cropping `girlfriend.png` may fix it - she's got a lot of empty space beneath her. I'm not sure what would need to be refactored in the code to compensate, but I'll try that experiment as well.

It seems as if there is a texture limit, the limit being 4096x4096 as it turns out

**Comment by AM7999** [comment]
If anyone in this issue thread has a Pi5 could they test it on that

**Comment by AM7999** [comment]
On the pi4 and pi5 i heard Gman say it was 4096x4096 @GuglioIsStupid, now i'm not sure about other pis

**Comment by GuglioIsStupid** [comment]
2048x2048 or 4096x4096, its either or

i just went based off a simple search result as I dont have my pi4 setup correctly, but i think it indeed is 4096x4096 for pi4/pi5

**Comment by DarkBlueStealth** [comment]
@GuglioIsStupid, I have both a pi3 and a pi4 on hand right now. I'll see what max textures they can handle just to verify.
